draft-irtf-cfrg-opaque.md

```diff
@@ -384,139 +383,223 @@ variant) of the OPRF described in {{I-D.irtf-cfrg-voprf}}.
+
+~~~
+    creds                             (parameters, record)
+      |                                         |
+      v                                         v
+    Client                                    Server
+    ------------------------------------------------
+                   AKE message 1
+             ------------------------->
+                   AKE message 2
+             <-------------------------
+                   AKE message 3
+             ------------------------->
+   ------------------------------------------------
+      |                                         |
+      v                                         v
+(export_key, session_key)                  session_key
+~~~
+
+The rest of this document describes the details of these stages in detail.
+{{client-credential-storage}} describes how client credential information is
+encoded, encrypted, and stored on the server. {{offline-phase}} describes the
+first registration stage of the protocol, and {{online-phase}} describes the
+second authentication stage of the protocol. {{configurations}} describes how
+to instantiate OPAQUE using different cryptographic dependencies and parameters.
+
+# Client Credential Storage {#client-credential-storage}
 
-OPAQUE allows applications to either provide custom client private and public keys for authentication or
-to generate them internally, making the application oblivious to the client's private key. Each public
-and private key value is an opaque byte string, specific to the AKE protocol in which OPAQUE is instantiated.
+OPAQUE makes use of a structure `Envelope` to manage client credentials.
+This envelope holds information about its format and content for the client to
+obtain its authentication material.
 
-These two options are defined as the `internal` and `external` modes.  See {{envelope-modes}} for their
-specifications.
+OPAQUE allows applications to either provide custom client private and public keys
+for authentication or to generate them internally, making the application oblivious
+to the client's private key. Each public and private key value is an opaque byte
+string, specific to the AKE protocol in which OPAQUE is instantiated.
+These two options are defined as the `internal` and `external` modes.
+See {{envelope-modes}} for their specifications.
 
-Applications may pin key material to identities if desired. If no identity is given for a party,
-its value MUST default to its public key. The following types of application credential information
-are hereby considered:
+Applications may pin key material to identities if desired. If no identity is given
+for a party, its value MUST default to its public key. The following types of
+application credential information are considered:
 
 - client_private_key: The encoded client private key for the AKE protocol.
 - client_public_key: The encoded client public key for the AKE protocol.
 - server_public_key: The encoded server public key for the AKE protocol.
```

Comment:
If we mention the client's private key in this part, it makes sense to also mention the server's private key. I suggest either both or none (as they are not used as identities).

Reply:
I mean the private keys

Reply:
I understand, but since the server private key isn't used for credentials, it's not included.